src/rules/semantic-function-prefixes.ts

```
@@ -62,16 +62,42 @@ export const semanticFunctionPrefixes = createRule<[], MessageIds>({
+      // - PascalCase: GetData -> Get
+      // - UPPERCASE: UPDATEUser -> UPDATE
+      // - Mixed: XMLHttpRequest -> XML
+
       for (let i = 1; i < methodName.length; i++) {
-        if (methodName[i] >= 'A' && methodName[i] <= 'Z') {
+        const currentChar = methodName[i];
+        const prevChar = methodName[i - 1];
+
+        // Case 1: lowercase followed by uppercase (camelCase boundary)
+        // Example: getData -> break at 'D'
+        if (prevChar >= 'a' && prevChar <= 'z' && currentChar >= 'A' && currentChar <= 'Z') {
```

ESLint

[error] Replace `prevChar·>=·'a'·&&·prevChar·<=·'z'·&&·currentChar·>=·'A'·&&·currentChar·<=·'Z'` with `⏎··········prevChar·>=·'a'·&&⏎··········prevChar·<=·'z'·&&⏎··········currentChar·>=·'A'·&&⏎··········currentChar·<=·'Z'⏎········`

(prettier/prettier)